### DISCUSSION TITLE
Post ID is too obsecure and long

### DISCUSSION THREAD

**Discussion by PressJump** [kamp-us/pano#448] [Brain storming - Fikir Paylaşımı]
### Describe the feature you would like to see added

Currently, the id system of pano is too complex for human-readable url's and causes the URL of a post to take up an excessive amount of text space. This can be seen from a example post made on Pano.

    posts/Slug-CUID

### Is your feature request related to a problem?

Current post URLs are too cryptic, lengthy, and suspicious-appearing when content is shared online as CUID's are very long for post identification purposes.

### Describe the solution you'd like

Currently, Reddit employs a six-character post id for each post/comment `www.reddit.com/comments/######`, which results in a human-readable slug that optimizes search engine results that are relevant to the search. This is seen in a typical Reddit post, when the entire post title is used as the slug after the post/comment id. 

`/comments/######/post-title-slug-here`

Utilizing a strategy like Reddit's post routing system will not only make it easier to discern connections and make it less suspicious appearing with a shorter id followed up with the slug.

### Describe alternatives you have considered

This example demonstrates how HackerNews utilises an earlier method of post routing including a URL parameter.

    https://news.ycombinator.com/item?id=########

Although this is a shorter alternative than the other proposed method, it does not provide a descriptive slug to distinguish postings which is not preferable.

**Comment by PressJump** [comment]
After talks this issue on a new id system was wanted to be applied on to post routes including comments.

Current planned way of implementing this is turning auto increment id to base36 and applying an offset of 100000. This will give us a 6 digit identifier that is better than CUID in terms of shorter routes like Reddit.

One problem with this type of identification is generating the identifier each time on the application level on a new post or a comment. Due to this it was decided that a lower level implementation is essential for this type of job.

For this, utilising mongodb functions through setup migrations will be a good way to implement this with minimal changes to the core in which this issue will not block any other issues/pull-requests.

Currently I am on works of providing a prototype of this working for a reassessment of the issue.

**Comment by usirin** [comment]
@PressJump any updates? should we wait for this?
